store/task.go

```diff
@@ -745,9 +755,13 @@ func (s *Store) patchTaskStatusImpl(ctx context.Context, tx *sql.Tx, patch *api.
 			return nil, FormatError(err)
 		}
 		taskPatchedRaw = &taskRaw
+		found = true
 	}
-	if err := row.Close(); err != nil {
-		return nil, err
+	if err := row.Err(); err != nil {
+		return nil, FormatError(err)
+	}
+	if !found {
```

Comment:
```suggestion
	if err := row.Close(); err != nil {
		return nil, err
	}
	if !found {
```

Reply:
With this change, all tests pass successfully.

Reply:
We cannot do this. What if `row.Scan` returns err? Then `row.Close()` will not be called.

Reply:
We can call `row.Close` multiple times.

Reply:
Done.